packages/ui/v2/core/Shell.tsx

```diff
@@ -906,7 +884,7 @@ function TopNav() {
     <>
       <nav
         style={isEmbed ? { display: "none" } : {}}
-        className="fixed z-40 flex w-full items-center justify-between border-b border-gray-200 bg-gray-50 bg-opacity-50 py-1.5 px-4 backdrop-blur-lg sm:relative sm:p-4 md:hidden">
+        className="sticky top-0 z-40 flex w-full items-center justify-between border-b border-gray-200 bg-gray-50 bg-opacity-50 py-1.5 px-4 backdrop-blur-lg sm:p-4 md:hidden">
```

Comment:
No fixed positioning anymore, sticky takes way better care of the job — sticking the nav to the top when the user scrolls down, but not sticking it when there's a topbar above the navigation,  until that is scrolled out of view.